GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -78,19 +78,20 @@ private void TimerCallback(object unusedState)
                         this.tracer,
                         this.fileSystem);
 
-                    // The upgrade check always goes against GitHub
-                    GitHubUpgrader productUpgrader = GitHubUpgrader.Create(
+                    ProductUpgrader.TryCreateUpgrader(
                         this.tracer,
                         this.fileSystem,
                         new LocalGVFSConfig(),
+                        credentialStore: null,
                         dryRun: false,
                         noVerify: false,
+                        newUpgrader: out ProductUpgrader productUpgrader,
                         error: out errorMessage);
 
                     if (productUpgrader == null)
                     {
                         string message = string.Format(
-                            "{0}.{1}: GitHubUpgrader.Create failed to create upgrader: {2}",
+                            "{0}.{1}: failed to create upgrader: {2}",
                             nameof(ProductUpgradeTimer),
                             nameof(this.TimerCallback),
                             errorMessage);

```

Comment:
All of these formats using `nameof(ProductUpgradeTimer), nameof(this.TimerCallback)` make me think we need a helper method to contruct this message.  They are all very similar.  Maybe we should be using the `EventMetadata` and message should be something simple like "Failed to create upgrader"?

Reply:
I like this suggestion - I might hold off on making this change as this code should change at some point (as we transition to deprecate the existing nuget upgrader logic)